scripts/license.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# license.sh checks that all Go files in the given correct-looking license header.
+
+check_header() {
+    got=$1
+    want=$(cat <<EOF
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+EOF
+    )
+    if [[ "$got" = *"$want"* ]]; then
+        return 0
+    fi
+    return 1
+}
+
+if [ $# != 1 ]; then
+    echo "Usage: $0 rootdir" >&2
+    exit 1
+fi
+
+fail=0
+for file in $(find -E $1 -regex '.*\.(go|js)'); do
+    case $file in
+        $1/*/mocks/*)
+            # Skip mocks
+        ;;
+        $1/*/node_modules/*)
+            # Skip node modules for js files.
+        ;;
+        $1/site/*)
+            # Skip website content
+        ;;
+        *)
+            header="$(head -10 $file)"
+            if ! check_header "$header"; then
```

Comment:
Sorry, forgive my bash ignorance--why is this a different syntax than [L48](https://github.com/aws/copilot-cli/pull/1310/files/5a8e821eb61d4a11ff1ffc64a581415252f79767..3974fbfbc373d5c03468f0049400713c115c03ae#diff-3016f5e81e1b55c1248aa79ba8a929e8R48)?

Reply:
I guess `! check_header` means if the function doesn't exit successfully instead of directly comparing the return value?

Reply:
I don't know anything about bash either :P

From my understanding: https://stackoverflow.com/a/47876317/1201381 the first one runs `check_header` so i can't use the test command `[` for it